sdk/test-utils/recorder/src/utils.ts

```diff
@@ -71,12 +71,19 @@ function padStart(currentString: string, targetLength: number, padString: string
 /**
  * @returns {string}
  */
+const namesAlreadyTaken: { [key: string]: boolean } = {};
 export function getUniqueName(prefix: string): string {
-  return `${prefix}${new Date().getTime()}${padStart(
+  const name = `${prefix}${new Date().getTime()}${padStart(
     Math.floor(Math.random() * 10000).toString(),
     5,
     "00000"
   )}`;
+  if (namesAlreadyTaken[name]) {
+    throw new Error(`Test name: ${name} is duplicated.`);
```

Comment:
The error might be a bit misleading.

![image](https://user-images.githubusercontent.com/10452642/70138211-9ca70480-1644-11ea-81cb-b417808c173d.png)

Also, it's better to have this `namesAlreadyTaken` at the `getUniqueName` function which is exposed from the `Recorder` object.
https://github.com/Azure/azure-sdk-for-js/blob/72bb17fa22d8426555b78e1372dc7ae658ff20a9/sdk/test-utils/recorder/src/recorder.ts#L126
And throw an error saying.. `Label <label> is already taken for the corresponding prefix, please provide a different prefix <prefix> OR give the same prefix with a new label.`

Reply:
Oh! I see. I pushed something that seems to tackle this issue better. Let me know if this works.

Reply:
No, the new code misses the case where two different prefixes are trying to get the label. According to the recorder logic, it is not allowed(meaning playback would fail).

The older `namesAlreadyTaken` was fine, but put that here and use it for the `label`(labels cannot be repeated) - https://github.com/Azure/azure-sdk-for-js/blob/72bb17fa22d8426555b78e1372dc7ae658ff20a9/sdk/test-utils/recorder/src/recorder.ts#L126
Is that clear?

Reply:
Pushed a commit to solve the issue without the need for a new array/set.